### PR TITLE
[rest] GA in docs!

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 1]
```diff
@@ -193,106 +194,264 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
-        self.files = files
-        self.data = data
+        self.method = method
+        self.headers = CaseInsensitiveDict(default_headers)
+
+    @property
+    def content(self) -> Any:
+        """Get's the request's content
+
+        :return: The request's content
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:225]: It is already showed in def content(self) -> Any:.

For dev doc, I don't think we need to duplicate the information?

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 2]
```diff
@@ -193,106 +194,141 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
-        """
-
-    def set_bytes_body(self, data):
-        """Set generic bytes as the body of the request."""
-
-    def set_multipart_mixed(self, *requests, **kwargs):
-        """Set requests for a multipart/mixed body.
-        Optionally apply "policies" in kwargs to each request.
-        """
+    def content(self) -> Any:
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:222]: Sorry I did not clarify enough.

What I meant was we only needed to introduce what the method/property does, no need to have the types.

We still want:

def content(self) -> Any:
"""Get's the request's content"""

Apologize for bringing the confusion.

Comment (iscai-msft) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:222]: ah ok, so we still want the sphinx documentation, we only don't needd the abstract method decorators?

Comment (xiangyan99) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:222]: e.g.:

def iter_raw(self, **kwargs: Any) -> Iterator[bytes]:
"""Iterates over the response's bytes. Will not decompress in the process."""

:)

Comment (iscai-msft) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:222]: ok, have updated to that and gotten rid of dunder methods. lmk!

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 3]
```diff
@@ -193,106 +194,215 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        ...
+
+    def close(self) -> None:
+        ...
+
+    def read(self) -> bytes:
+        ...
+
+    def iter_raw(self, **kwargs: Any) -> Iterator[bytes]:
+        ...
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:368]: Is there any reason we only have introduction for async but not for sync?

Comment (iscai-msft) [reply sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:368]: oh let me fix that

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 4]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        If response is good, does nothing.
+
+        :rtype: None
+        :raises ~azure.core.HttpResponseError if the object has an error status code.:
+        """
+        ...
 
-    def set_json_body(self, data):
-        """Set a JSON-friendly object as the body of the request."""
+    def close(self) -> None:
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:361]:
```suggestion

```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 5]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        ...
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:362]:
```suggestion

```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 6]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+    def content_type(self) -> Optional[str]:
+        """The content type of the response.
+
+        :rtype: str
+        """
+        ...
+
+    @property
+    def is_closed(self) -> bool:
+        """Whether the network connection has been closed yet.
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:284]:
```suggestion
        """Whether the network connection has been closed yet."""
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 7]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+
+        :rtype: bool
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:286]:
```suggestion
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 8]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        """
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:287]:
```suggestion
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 9]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        ...
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:288]:
```suggestion
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 10]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
         self.url = url
-        self.headers = CaseInsensitiveDict(headers)
-        self.files = files
-        self.data = data
+        self.method = method
+        self.headers = CaseInsensitiveDict(default_headers)
+
+    @property
+    def content(self) -> Any:
+        """Get's the request's content
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:223]:
```suggestion
        """Get's the request's content"""
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 11]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+
+        :return: The request's content
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:225]:
```suggestion
```

File: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md [hunk 12]
```diff
@@ -193,106 +194,230 @@ proxy_policy.proxies = {'https': 'http://user:password@10.10.1.10:1180/'}
...
+        :rtype: any
```
Comment (xiangyan99) [inline sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md:226]:
```suggestion
```

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**